packages/rx_bloc_cli/mason_templates/bricks/feature_google_login/__brick__/lib/lib_auth_google/services/google_login_service.dart

```diff
@@ -0,0 +1,63 @@
+import 'dart:developer';
+
+import 'package:google_sign_in/google_sign_in.dart';
+
+import '../../base/app/config/app_constants.dart';
+import '../../base/models/errors/error_model.dart';
+import '../../base/repositories/push_notification_repository.dart';
+import '../../lib_auth/repositories/auth_repository.dart';
+import '../../lib_permissions/services/permissions_service.dart';
+import '../repositories/google_auth_repository.dart';
+
+class GoogleLoginService {
+  GoogleLoginService(
+    this._googleAuthRepository,
+    this._permissionsService,
+    this._authRepository,
+    this._pushSubscriptionRepository,
+    this._googleSignIn,
+  );
+  final GoogleAuthRepository _googleAuthRepository;
+  final PermissionsService _permissionsService;
+  final AuthRepository _authRepository;
+  final PushNotificationRepository _pushSubscriptionRepository;
+  final GoogleSignIn _googleSignIn;
+
+  Future<bool> googleLogin() async {
```

Comment:
Please build an abstraction, which avoid copy/paste among all auth services. Polymorphism could be a good choice :)

Reply:
Will do